openresty-build-tools/kong-ngx-build

```diff
@@ -321,6 +337,11 @@ main() {
     fi
   popd
 
+  if [ ! -z "$DOWNLOAD_ONLY" ]; then
+    notice "Exiting early since --download-extract-only was specified"
+    exit 0
+  fi
+
```

Comment:
should the early exit happen after applying patches?

Reply:
I did debate on where to put this escape hatch.

The patching isn't gated by the existence of the extracted download directories the same way the download & extraction is. `patch` is pretty forgiving when you ask it to re-apply a patch that is already in place too. So I'm unconcerned about pushing the patching to subsequent runs.